pkg/tfbridge/info/rename.go

```diff
@@ -84,7 +84,8 @@ func (p *Provider) RenameDataSource(resourceName string, legacyTok tokens.Module
 			currentInfo.Tok.Name().String()))
 	p.DataSources[resourceName] = &currentInfo
 	p.DataSources[legacyResourceName] = &legacyInfo
-	p.P.DataSourcesMap().Set(legacyResourceName, p.P.DataSourcesMap().Get(resourceName))
+	err := shim.CloneResource(p.P.DataSourcesMap(), resourceName, legacyResourceName)
```

Comment:
Now would be a great opportunity to add a comment to `RenameDataSource` that explains what it actually does (and fix `resourceName` to `datasourceName`).